pennylane/fermi/fermionic.py

```diff
@@ -242,3 +243,44 @@ def simplify(self, tol=1e-8):
         for fw, coeff in items:
             if abs(coeff) <= tol:
                 del self[fw]
+
+
+def string_to_fermi_word(fermi_string):
+    r"""Return a fermionic operator object from its string representation.
+
+    The string representation is a compact format that uses the orbital index and `+` or `-` symbols
+    to indicate creation and annihilation operators, respectively. For instance, the string
+    representation for the operator :math:`a\dagger_0 a_1 a\dagger_0 a_1` is '0+ 1- 0+ 1-'. The `-`
+    symbols can be optionally dropped such that '0+ 1 0+ 1' represents the same operator. The format
+    commonly used in OpenFermion, '0^ 1 0^ 1' to represent the same operator, is also supported.
+
+    Args:
+        fermi_string (str): string representation of the fermionic object
+
+    Returns:
+        FermiWord: the fermionic operator object
+
+    **Example**
+
+    >>> string_to_fermi_word('0+ 1- 0+ 1-')
+    <FermiWord = '0+ 1- 0+ 1-'>
+
+    >>> string_to_fermi_word('0+ 1 0+ 1')
+    <FermiWord = '0+ 1- 0+ 1-'>
+
+    >>> string_to_fermi_word('0^ 1 0^ 1')
+    <FermiWord = '0+ 1- 0+ 1-'>
+    """
+    if fermi_string.isspace() or not fermi_string:
+        return FermiWord({})
+
+    fermi_string = re.sub(" +", " ", fermi_string).strip()
+
+    if not all(s.isdigit() or s in ["+", "-", "^", " "] for s in fermi_string):
+        raise ValueError(f"Invalid character encountered in string {fermi_string}.")
```

Comment:
In terms of input validation, do we also want to check that no one does this: 

`string_to_fermi_word("+0 -1 +0 -1")`

Doesn't seem like a completely unreasonable mistake, but we did tell them the format in the docstring 🤷‍♀️